File: chat-proxy/src/main/java/org/example/domain/chat/adapter/ChatListener.java
package org.example.domain.chat.adapter;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.example.JsonConverter;
import org.example.domain.ChatMessage;
import org.springframework.kafka.annotation.KafkaListener;
import org.springframework.stereotype.Component;

@Component
@Slf4j
@RequiredArgsConstructor
public class ChatListener {


    private final JsonConverter jsonConverter;

    @KafkaListener(topics ="${kafka.topic.chat}",groupId = "${kafka.group.chat}")
    public void resultTaskListener(String chatMessage){
        ChatMessage message = null;
        try{
            message = jsonConverter.toObject(chatMessage, ChatMessage.class);
        } catch (Exception e){

        } finally {

        }
    }
}

File: proxy-kotlin/src/main/kotlin/com/chat/proxykotlin/domain/chat/ChatListener.kt
package com.chat.proxykotlin.domain.chat

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.module.kotlin.readValue
import org.slf4j.LoggerFactory
import org.springframework.kafka.annotation.KafkaListener

import org.springframework.stereotype.Component

@Component
class ChatListener(
    private val objectMapper: ObjectMapper,
) {

    private val log = LoggerFactory.getLogger(ChatListener::class.java)


    @KafkaListener(topics = ["\${spring.kafka.topic.chat}"], groupId = "1")
    fun chatConsume(message: String){
        log.info(message)

        val chatMessage = objectMapper.readValue(message,ChatMessage::class.java)
    }


}